pyquil/paulis.py

```diff
@@ -574,7 +577,7 @@ def __mul__(self, other):
         new_sum = PauliSum(new_terms)
         return new_sum.simplify()
 
-    def __rmul__(self, other):
+    def __rmul__(self, other: Number) -> 'PauliSum':
```

Comment:
use of `Number`

Reply:
Do I change all use of `Number` to `ExpressionValueDesignator`?

Reply:
Sounds like @appleby laid out our options above (I was just pointing out the discrepancy). I'll respond in the other thread.

Reply:
I guess change this to `PauliCoefficientDesignator` for consistency with other `PauliSum` arithmetic ops.